app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/DownloadAttachmentProgressDialog.kt

```diff
@@ -76,8 +79,14 @@ class DownloadAttachmentProgressDialog : DialogFragment() {
     }
 
     private fun popBackStackWithError() {
-        val title = if (mainViewModel.isInternetAvailable.value == true) R.string.anErrorHasOccurred else R.string.noConnection
-        showSnackbar(title)
-        findNavController().popBackStack()
+        // We have to use the Flow instead of the LiveData here since the mainViewModel is not the same instance as the one in the
+        // MainActivity. So, instead of observing the LiveData in order to make it "active", we use the Flow.
+        lifecycleScope.launch {
```

Comment:
Can you please specify the Dispatcher ?

Reply:
The `Dispatchers` is not important cause we're already in the main Dispatchers, but it's the wrong scope, use `viewLifecycleOwner.lifecycleScope` instead